### PR TITLE
Add typography

### PR DESCRIPTION
**Pull request by maya**
The adds the typography section of the library, as well as correct typography throughout the site.

Adds the typeface intro, style pairings, typesetting, links, and lists.

### CODE REVIEW

File: assets/_scss/elements/_typography.scss [hunk 1]
```diff
@@ -175,3 +177,15 @@ dfn {
 .usa-drop_text {
   margin-bottom: 0;
 }
+
+.usa-background-dark {
+  background-color: #333B46;
+
+  p {
```
Comment (msecret) [inline assets/_scss/elements/_typography.scss]: What about span, or form elements?

File: assets/_scss/elements/_typography.scss [hunk 2]
```diff
@@ -175,3 +177,15 @@ dfn {
 .usa-drop_text {
   margin-bottom: 0;
 }
+
+.usa-background-dark {
+  background-color: #333B46;
```
Comment (msecret) [inline assets/_scss/elements/_typography.scss]: Why is this not a variable?

Comment (maya) [inline assets/_scss/elements/_typography.scss]: The color was not one of our variables, but I'll check with @mollieru

File: assets/_scss/elements/_typography.scss (hunk 1 above)
Comment (maya) [inline assets/_scss/elements/_typography.scss]: I can add span too, which form elements?

File: assets/_scss/elements/_typography.scss (hunk 2 above)
Comment (maya) [inline assets/_scss/elements/_typography.scss]: Confirmed with @angel to use `$color-gray-dark` - done in 717868c.

File: assets/_scss/elements/_typography.scss (hunk 1 above)
Comment (maya) [inline assets/_scss/elements/_typography.scss]: @msecret I added span in aea5292. If you let me know which form elements you think should be added, then I can add it as well.

**Comment by maya** [comment]
Font guidance will need updating, but let's review as is and get the ball rolling so others can use the newest typography. @msecret

File: assets/css/styleguide.scss [hunk 3]
```diff
@@ -858,3 +883,126 @@ code[class*="language-"], pre[class*="language-"] {
...
+  .usa-background-dark {
+    display: inline-block;
+    padding: {
+      left: 1em;
+      right: 1em;
+    }
+  }
+}
+
+.usa-link-visited {
```
Comment (msecret) [inline assets/css/styleguide.scss]: Why would this rule be here? Isn't it in the pattern library?

File: assets/css/styleguide.scss [hunk 4]
```diff
@@ -858,3 +883,126 @@ code[class*="language-"], pre[class*="language-"] {
...
+    font-family: $font-serif;
+    font-size: $h3-font-size;
+  }
+
+  .usa-font-lead-alt {
+    font-size: $h5-font-size;
+  }
+}
+
+.sans-style {
```
Comment (msecret) [inline assets/css/styleguide.scss]: Is this a pattern library style? Why are these sizes being modified? Shouldn't we be showing the users the exact font sizes they should be getting when they use the library?

**Comment by msecret** [comment]
How is the performance.png being used? The image stays at the same level of performance so how is it showing different performance?

**Comment by maya** [comment]
@msecret re: the performance.png - vizdee is working on custom graphics for that and what's there now is placeholder.

File: assets/css/styleguide.scss (hunk 4 above)
Comment (maya) [inline assets/css/styleguide.scss]: @msecret the reason why font sizes are modified is bc there are different sizes when serif and sans serifs are the base fonts. 

I think the correct way to do this is provide a different typography.scss and variables.scss depending on which family they choose. This was just done for now and to get it up quickly, but let me know if you think there's a better way to do this.

File: assets/css/styleguide.scss (hunk 3 above)
Comment (maya) [inline assets/css/styleguide.scss]: @msecret are you saying we should add this class along with `a:visited` in typography.scss?

Comment (msecret) [inline assets/css/styleguide.scss]: Yea, if it's being done here right? Is there a reason this should only be in the styleguide?

File: assets/css/styleguide.scss (hunk 4 above)
Comment (maya) [inline assets/css/styleguide.scss]: Or we can provide the correct values in the documentation for each family?

File: assets/css/styleguide.scss (hunk 3 above)
Comment (maya) [inline assets/css/styleguide.scss]: It brings into question how many color helpers we want to provide. For example, developers can add this background color in however they want. 

These are all the colored backgrounds we are showing as possible background colors to use:

screen shot 2015-08-31 at 9 53 28 am

The last thing we decided was to _not_ provide the color helpers.

File: assets/css/styleguide.scss (hunk 4 above)
Comment (msecret) [inline assets/css/styleguide.scss]: Yea that could get complicated for the user to set up but I understand what you're doing now.

File: assets/css/styleguide.scss (hunk 3 above)
Comment (maya) [inline assets/css/styleguide.scss]: oops, I thought this was a different question

Comment (maya) [inline assets/css/styleguide.scss]: Done in: dff445b.

File: assets/css/styleguide.scss [hunk 5]
```diff
@@ -858,3 +883,122 @@ code[class*="language-"], pre[class*="language-"] {
...
+
+  h5 {
+    font-size: rem(16px);
+  }
+
+  h6 {
+    font-size: rem(13px);
+  }
+
+  .usa-display {
```
Comment (msecret) [inline assets/css/styleguide.scss]: Why wouldn't display and lead be in the pattern library?

Comment (maya) [inline assets/css/styleguide.scss]: @msecret they are, on line 143-158 of typography.scss - these are the unique "other font pairings" as well as "robust" style with the `usa-lead-alt`.

**Pull request merged**